packages/appium/lib/appium.js

```diff
@@ -201,10 +201,29 @@ class AppiumDriver extends DriverCore {
       'Please check https://github.com/appium/appium/issues/20880 for more details.');
   }
 
-  async getAppiumSessionCapabilities () {
-    throw new errors.NotImplementedError('TODO: Implement this method body to call getAppiumSessionCapabilities ' +
-      'from the destination driver instance in a session.'
-    );
+
+  /**
+   * Return session capabilities info related to the destination driver.
+   * @param sessionId - the id of the session to return capabilities info.
+   * @returns {Promise<{value: AppiumSessionCapabilities, error?: Error, protocol: string|undefined}>}
+   */
+  async getAppiumSessionCapabilities (sessionId) {
+    const dstSession = this.sessions[sessionId];
+    if (!dstSession) {
+      throw new Error(`The session with id '${sessionId}' does not exist`);
+    }
+
+    // to extend the newCommandTimeout for session id related endpoint call.
+    await dstSession.startNewCommandTimeout();
+
+    if (!_.isFunction(dstSession.getAppiumSessionCapabilities)) {
+      throw new errors.UnsupportedOperationError('The driver does not support the endpoint.');
```

Comment:
We could improve this error message: `${dstSession.constructor.name} does not support session capabilities retrieval yet`